### PR TITLE
test: fix unstabled drop then add column

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 

fix https://github.com/pingcap/dm/pull/1540#issuecomment-809451583

### What is changed and how it works?

move add column after a sync-diff check

### Check List 

Tests 

 - No code

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by lance6716** [comment]
another unstable test, maybe I should fix it in this PR https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/8648/pipeline/87

**Comment by lance6716** [comment]
@lichunzhu the log is not complete 🤔 

```
[2021-03-30T04:03:43.285Z] [2021/03/30 12:03:12.963 +08:00] [DEBUG] [optimist.go:476] ["lock for info not found"] [component="shard DDL optimist"] [info="{\"task\":\"test\",\"source\":\"mysql-replica-01\",\"up-schema\":\"shardddl1\",\"up-table\":\"tb1\",\"down-schema\":\"shardddl\",\"down-table\":\"tb\",\"ddls\":[\"ALTER TABLE `shardddl`.`tb` ADD COLUMN `b` INT AFTER `a`\"],\"table-info-before\":{\"id\":54,\"name\":{\"O\":\"tb1\",\"L\":\"tb1\"},\"charset\":\"latin1\",\"collate\":\"latin1_bin\",\"cols\":[{\"id\":1,\"name\":{\"O\":\"a\",\"L\":\"a\"},\"offset\":0,\"origin_default\":null,\"origin_default_bit\":null,\"default\":null,\"default_bit\":null,\"default_is_expr\":false,\"generated_expr_string\":\"\",\"generated_stored\":false,\"dependences\":null,\"type\":{\"Tp\":3,\"Flag\":4099,\"Flen\":11,\"Decimal\":0,\"Charset\":\"binary\",\"Collate\":\"binary\",\"Elems\":null},\"state\":5,\"comment\":\"\",\"hidden\":false,\"change_state_info\":null,\"version\":2},{\"id\":4,\"name\":{\"O\":\"c\",\"L\":\"c\"},\"offset\":1,\"origin_default\":null,\"origin_default_bit\":null,\"default\":null,\"default_bit\":null,\"default_is_expr\":false,\"generated_expr_string\":\"\",\"generated_stored\":false,\"dependences\":null,\"type\":{\"Tp\":3,\"Flag\":0,\"Flen\":11,\"Decimal\":0,\"Charset\":\"binary\",\"Collate\":\"binary\",\"Elems\":null},\"state\":5,\"comment\":\"\",\"hidden\":false,\"change_state_info\":null,\"version\":2}],\"index_info\":null,\"constraint_info\":null,\"fk_info\":null,\"state\":5,\"pk_is_handle\":true,\"is_common_handle\":false,\"common_handle_version\":0,\"comment\":\"\",\"auto_inc_id\":0,\"auto_id_cache\":0,\"auto_rand_id\":0,\"max_col_id\":4,\"max_idx_id\":0,\"max_cst_id\":0,\"update_timestamp\":423907028164935680,\"ShardRowIDBits\":0,\"max_shard_row_id_bits\":0,\"auto_random_bits\":0,\"pre_split_regions\":0,\"partition\":null,\"compression\":\"\",\"view\":null,\"sequence\":null,\"Lock\":null,\"version\":4,\"tiflash_replica\":null,\"is_columnar\":false},\"table-info-after\":[{\"id\":54,\"name\":{\"O\":\"tb1\",\"L\":\"tb1\"},\"charset\":\"latin1\",\"collate\":\"latin1_bin\",\"cols\":[{\"id\":1,\"name\":{\"O\":\"a\",\"L\":\"a\"},\"offset\":0,\"origin_default\":null,\"origin_default_bit\":null,\"default\":null,\"default_bit\":null,\"default_is_expr\":false,\"generated_expr_string\":\"\",\"generated_stored\":false,\"dependences\":null,\"type\":{\"Tp\":3,\"Flag\":4099,\"Flen\":11,\"Decimal\":0,\"Charset\":\"binary\",\"Collate\":\"binary\",\"Elems\":null},\"state\":5,\"comment\":\"\",\"hidden\":false,\"change_state_info\":null,\"version\":2},{\"id\":5,\"name\":{\"O\":\"b\",\"L\":\"b\"},\"offset\":1,\"origin_default\":null,\"origin_default_bit\":null,\"default\":null,\"default_bit\":null,\"default_is_expr\":false,\"generated_expr_string\":\"\",\"generated_stored\":false,\"dependences\":null,\"type\":{\"Tp\":3,\"Flag\":0,\"Flen\":11,\"Decimal\":0,\"Charset\":\"binary\",\"Collate\":\"binary\",\"Elems\":null},\"state\":5,\"comment\":\"\",\"hidden\":false,\"change_state_info\":null,\"version\":2},{\"id\":4,\"name\":{\"O\":\"c\",\"L\":\"c\"},\"offset\":2,\"origin_default\":null,\"origin_default_bit\":null,\"default\":null,\"default_bit\":null,\"default_is_expr\":false,\"generated_expr_string\":\"\",\"generated_stored\":false,\"dependences\":null,\"type\":{\"Tp\":3,\"Flag\":0,\"Flen\":11,\"Decimal\":0,\"Charset\":\"binary\",\"Collate\":\"binary\",\"Elems\":null},\"state\":5,\"comment\":\"\",\"hidden\":false,\"change_state_info\":null,\"version\":2}],\"index_info\":null,\"constraint_info\":null,\"fk_info\":null,\"state\":5,\"pk_is_handle\":true,\"is_common_handle\":false,\"common_handle_version\":0,\"comment\":\"\",\"auto_inc_id\":0,\"auto_id_cache\":0,\"auto_rand_id\":0,\"max_col_id\":5,\"max_idx_id\":0,\"max_cst_id\":0,\"update_timestamp\":423907028200849408,\"ShardRowIDBits\":0,\"max_shard_row_id_bits\":0,\"auto_random_bits\":0,\"pre_split_regions\":0,\"partition\":null,\"compression\":\"\",\"view\":null,\"sequence\":null,\"Lock\":null,\"version\":4,\"tiflash_replica\":null,\"is_columnar\":false}],\"ignore-conflict\":false}"]
```

lock for ADD of drop-add-column DDL not found, so not detected drop-add-column

**Comment by lance6716** [comment]
there's no `mysql-replica-02`

between

```
[2021/03/30 12:03:15.429 +08:00] [INFO] [lock.go:525] ["table added to the lock"] [lock=test-`shardddl`.`tb`] [source=mysql-replica-02] [schema=shardddl1] [table=tb1] ["table info"="CREATE TABLE `tbl`(`a` INT(11) NOT NULL, `b` INT(11), PRIMARY KEY (`a`)) CHARSET UTF8MB4 COLLATE UTF8MB4_0900_AI_CI"]
```

and 

```
[2021/03/30 12:03:21.025 +08:00] [INFO] [lock.go:525] ["table added to the lock"] [lock=test-`shardddl`.`tb`] [source=mysql-replica-02] [schema=shardddl1] [table=tb1] ["table info"="CREATE TABLE `tbl`(`a` INT(11) NOT NULL, PRIMARY KEY (`a`)) CHARSET UTF8MB4 COLLATE UTF8MB4_0900_AI_CI"]
```

🤔

File: tests/shardddl3/run.sh [hunk 1]
```diff
@@ -1083,9 +1083,9 @@ function DM_DropAddColumn() {
     do
     echo "run DM_DropAddColumn case #${i}"
     run_case DropAddColumn "double-source-optimistic" \
-    "run_sql_source1 \"create table ${shardddl1}.${tb1} (a int primary key, b int, c int);\"; \
-     run_sql_source2 \"create table ${shardddl1}.${tb1} (a int primary key, b int, c int);\"" \
-    "clean_table" "optimistic" "$i"
+        "run_sql_source1 \"create table ${shardddl1}.${tb1} (a int primary key, b int, c int);\"; \
+            run_sql_source2 \"create table ${shardddl1}.${tb1} (a int primary key, b int, c int);\"" \
+        "clean_table" "optimistic" "$i"
```
Comment (GMHDBJD) [inline tests/shardddl3/run.sh:1088]:
```suggestion
    "run_sql_source1 \"create table ${shardddl1}.${tb1} (a int primary key, b int, c int);\"; \
     run_sql_source2 \"create table ${shardddl1}.${tb1} (a int primary key, b int, c int);\"" \
    "clean_table" "optimistic" "$i"
```

**Review by GMHDBJD — COMMENTED** [review]
rest LGTM

**Review by GMHDBJD — APPROVED** [review]
/lgtm

**Pull request merged**